src/foam/u2/dialog/ConfirmationModal.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.dialog',
+  name: 'ConfirmationModal',
+  extends: 'foam.u2.dialog.StyledModal',
+  documentation: `
+    Extension of styled modal with a primary and secondary action, mainly to be used for conifrmations and yes/no modals.
+    Clicking on any action closes performs the action and closes the dialog
+  `,
+
+  imports: ['theme?'],
+
+  messages: [{ name: 'CancelLabel', message: 'Cancel' }],
+
+  properties: [
+    {
+      class: 'FObjectProperty',
+      of: 'foam.core.Action',
+      name: 'primaryAction',
```

Comment:
Should this be set to` required: true` because we do not have a fallback primary action?